src/Widgets/AppButton.vala

```diff
@@ -201,4 +200,26 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
         }
     }
 #endif
+
+    private bool create_context_menu (Gdk.Event e) {
+        menu = new Slingshot.AppContextMenu (desktop_id, desktop_path);
+        menu.app_launched.connect (() => {
+            app_launched ();
+        });
+
+        if (menu.get_children () != null) {
+            switch (e.type) {
```

Comment:
Since we're only checking for two cases, it would probably be more concise to just use else/if